### DISCUSSION TITLE
Close persistent notification automatically

### DISCUSSION THREAD

**Discussion by howdu** [filamentphp/filament#13410] [Help]
### Package

Notifications

### Package Version

v3.2.92

### How can we help you?

Is there a way of interacting with notifications on the page?

E.g want to remove this persistent notification automatically so that it's no longer shown when the page is save successfully.

```php
protected function beforeSave(): void
{
    if (! $this->customValidation()) {
        Notification::make('custom-error')
            ->danger()
            ->title('Test')
            ->persistent()
            ->send();

        $this->halt();
    }

    Notification::make('custom-error')->close();
}
```

**Comment by leandrocfe** [comment]
I don't understand why you are using persistent if you want to close it automatically.

**Reply by howdu** [reply]
E.g first time you click save it shows the error.... you fix the error in fields.... click save again it remove the error notification and shows the success.

I do not want to show the error notification when it's no longer invalid. Especially do not want to show the success and error notifications together.

**Reply by howdu** [reply]
Even without persistent and a delay if the user fixes the error and saves the page again it will show the error and success notifications together. It would be nice if there was a way of automatically removing a notification so this does not happen.

**Reply by leandrocfe** [reply]
Why not sending validation notifications?
https://filamentphp.com/docs/3.x/forms/validation#sending-validation-notifications

**Reply by howdu** [reply]
We're not looking to affect all the fields validation messages only this custom one.

I'm going to submit a PR to add an alpine event to close a notification.
